wrappers/gin/gin.go

```diff
@@ -12,18 +18,67 @@ import (
 // TracerKey is the key of the epsagon tracer in the gin.Context Keys map passed to the handlers
 const TracerKey = "EpsagonTracer"
 
+// EpsagonContext creates a context.Background() with epsagon's associated tracer for nexted instrumentations
+func EpsagonContext(c *gin.Context) context.Context {
+	return epsagon.ContextWithTracer(c.Keys[TracerKey].(tracer.Tracer))
+}
```

Comment:
making sure - a gin context is safe within a request scope, right? (And if its passed to some goroutine it has to be copied by the user using c.Copy())

Reply:
It is a pointer, so it is passed by reference usually. Also it is a struct, so by some aspects it is safe and others it is not.. but it is meant to be used for this purpose.

Reply:
my bad, I should have done more specific - I actually meant context.Keys which is a map. And I guess it still safe?

Reply:
Can you explain what "safe" means? it is a normal map, it won't stab anyone in the chest but it is also not using locks to manage writes and reads.
You can see the example application to see how it is used.